drracket/drracket/private/frame.rkt

```diff
@@ -670,16 +670,16 @@
               (semaphore-post wait-to-break) 
               (with-output-to-file tmp-filename 
                 (λ () 
-                  (let loop ([total 0]) 
+                  (let loop ([total 0])
                     (when gauge 
                       (send gauge set-value  
                             (inexact->exact 
-                             (floor (* 100 (/ total size)))))) 
-                    (let ([s (read-string 1024 port)]) 
+                             (floor (* 100 (/ total size))))))
+                    (define s (read-string 1024 port))
+                    (unless (eof-object? s) 
                       (unless (eof-object? s) 
-                        (unless (eof-object? s) 
-                          (display s) 
-                          (loop (+ total (string-length s))))))))
+                        (display s) 
```

Comment:
doubled `unless`! Guess that won't be a common error :)

Reply:
Yeah I can't imagine that one happening more than once or twice. Sometimes you just gotta be sure :p